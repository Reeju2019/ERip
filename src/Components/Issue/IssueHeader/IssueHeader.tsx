import React from 'react'
import { Container } from 'react-bootstrap'
import { Link, useParams } from 'react-router-dom'

const IssueHeader: React.FunctionComponent = () => {
  const { modelId } = useParams()
  const device: string = localStorage.getItem('deviceId') as string
  const brand: string = localStorage.getItem('brandId') as string
  const deviceId = JSON.parse(device)
  const brandId = JSON.parse(brand)
  return (
    <>
      <section className='bg-grey py-5'>
        <Container>
          <h3>Device Repair Details</h3>
          <div>
            <Link to='/' className='link-btn'>
              Home
              <i className='fa-solid fa-angle-right m-0' />
            </Link>
            <Link to={`/${deviceId}`} className='link-btn text-capitalize'>
              {deviceId} <i className='fa-solid fa-angle-right m-0' />
            </Link>
            <Link to={`/${deviceId}/${brandId}`} className='link-btn text-capitalize'>
              {brandId} <i className='fa-solid fa-angle-right m-0' />
            </Link>
            <Link to={``} className='link-btn active-link'>
              {modelId}
            </Link>
          </div>
        </Container>
      </section>
    </>
  )
}

export default IssueHeader
