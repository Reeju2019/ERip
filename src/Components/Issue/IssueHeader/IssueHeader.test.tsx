import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import IssueHeader from './IssueHeader'
import { MemoryRouter as Router } from 'react-router-dom'

test('issue Header', () => {
  render(
    <Router>
      <IssueHeader />
    </Router>,
  )
})
describe('App rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(
      <Router>
        <IssueHeader />
      </Router>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(
      <Router>
        <IssueHeader />
      </Router>,
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})
