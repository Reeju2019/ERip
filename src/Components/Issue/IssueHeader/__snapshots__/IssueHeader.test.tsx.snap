// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering specification SnapShot testing 1`] = `
<section
  className="bg-grey py-5"
>
  <div
    className="container"
  >
    <h3>
      Device Repair Details
    </h3>
    <div>
      <a
        className="link-btn"
        href="/"
        onClick={[Function]}
      >
        Home
        <i
          className="fa-solid fa-angle-right m-0"
        />
      </a>
      <a
        className="link-btn text-capitalize"
        href="/null"
        onClick={[Function]}
      >
         
        <i
          className="fa-solid fa-angle-right m-0"
        />
      </a>
      <a
        className="link-btn text-capitalize"
        href="/null/null"
        onClick={[Function]}
      >
         
        <i
          className="fa-solid fa-angle-right m-0"
        />
      </a>
      <a
        className="link-btn active-link"
        href="/"
        onClick={[Function]}
      />
    </div>
  </div>
</section>
`;
